### PR TITLE
Add data types and other minor cleanups

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by saz** [comment]
As Windows isn't supported by the module (and I think it's the same for the `gnupg` module, I think it's fine to change https://github.com/voxpupuli/puppet-rvm/blob/1949ca3aa8834e2b5c39331f6fe2bc53e00c8ae3/manifests/params.pp#L3

**Comment by ekohl** [comment]
I didn't see your comment, but I now got rid of Windows support. Also added more data types.

File: manifests/rvmrc.pp [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-  $manage_group = $rvm::params::manage_group,
+  Boolean $manage_group = $rvm::params::manage_group,
   $template = 'rvm/rvmrc.erb',
-  $umask = 'u=rwx,g=rwx,o=rx',
-  $max_time_flag = undef,
-  $autoupdate_flag = 0,
-$silence_path_mismatch_check_flag = undef) inherits rvm::params {
+  String[0] $umask = 'u=rwx,g=rwx,o=rx',
+  Optional[Integer[0]] $max_time_flag = undef,
+  Enum['disabled', 'warn', 'enabled'] $autoupdate_flag = 'disabled',
```
Comment (ekohl) [inline manifests/rvmrc.pp:7]: We can continue accepting `0`, `1` and `2` but I think just the named values provides a self-documenting experience.

Comment (saz) [reply manifests/rvmrc.pp:7]: Can't we add both ways, just as rvm is doing it?

Comment (ekohl) [reply manifests/rvmrc.pp:7]: We can, but it does make the data type more complicated. For example:
```suggestion
  Variant[Enum['disabled', 'warn', 'enabled'], Integer[0, 2]] $autoupdate_flag = 'disabled',
```

Or even:
```suggestion
  Variant[Enum['disabled', 'warn', 'enabled', '0', '1', '2'], Integer[0, 2]] $autoupdate_flag = 'disabled',
```

Personally I'm leaning to keeping it simple. The enum as it is in the commit now doesn't need documentation while the integers do.

Comment (saz) [reply manifests/rvmrc.pp:7]: Thanks, then let's go with simple, as there are already other breaking changes.

**Pull request merged**